src/node.cc

```diff
@@ -840,6 +842,26 @@ void SetupNextTick(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(ret);
 }
 
+void WeakCallback(
+    const v8::WeakCallbackInfo<v8::Persistent<v8::Promise,
+      node::ResetInDestructorPersistentTraits<v8::Promise> > >& data) {
+  Environment* env = Environment::GetCurrent(data.GetIsolate());
+  Persistent<Promise>* p = data.GetParameter();
+  Local<Promise> promise = p->Get(env->isolate());
+  Local<Value> err = promise->Result();
+  Persistent<Value>* e = new Persistent<Value>(env->isolate(), err);
+
+  env->SetImmediate([](Environment* env, void* data) {
+    Persistent<Value>* e = static_cast<Persistent<Value>*>(data);
+    Local<Value> err = PersistentToLocal(env->isolate(), *e);
+    Local<Message> message = Exception::CreateMessage(env->isolate(), err);
```

Comment:
@BridgeAR I think I mentioned that you need a `HandleScope` if you create new handles? Haven’t tried again so far, but that’s definitely something that made the process crash in debug mode here…